### ISSUE TITLE
Improvements needed in select facility

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#7330]
The select facility is still "work in progress".  I suggest certain improvements are needed, before the next version.

![image](https://user-images.githubusercontent.com/11226469/160272611-72c69455-23f8-4d6d-a1c9-f40d563d0e91.png)

I think @shadrackkibet is already working on the fact that calculations do not work yet once a selection is in operation.  
The calculate above shows another small problem.  I am not sure quite why these variables are not available in the dialogue.  They can be used, simply by typing, as shown.

Then, pressing Ok, gives an error.

I had forgotten that columns can be added with many other dialogues.  So I get the same problem with the Prepare > Data Frame > Row Names/Numbers dialogue.  I think this is the same thing, because it is adding a variable, so has to know where.

![image](https://user-images.githubusercontent.com/11226469/160272816-3001b995-52e4-4e3f-ac06-8e0473a03774.png)

I hope this can be sorted soon.

**Comment by lilyclements** [comment]
@rdstern this bug is occurring from the `add_columns_to_data` function.

Namely, in the line
``` private$data[[curr_col_name]] <- curr_col ```
(This is currently line 659, but may not be after future merges!)

The issue is that a new column is added (or, in the case of rearranging, is rearranged) in the unselected data set. But is not added to the selected data set.
This is because `private$data` adds a new column to the unselected "original" data set.

However, I am not sure of an alternative to `private$data` (`self$get_data_frame() <- self$get_data_frame() %>% mutate(curr_col_name = curr_col)` doesn't work, I guess that makes sense as to why we even have `add_columns_to_data` in the first place!).

This is one that goes a little over my head so might have to ask @dannyparsons for help (or @volloholic).

**Comment by Ivanluv** [comment]
> The problem here is now that I can see how to do it for the unselected data, but not sure how to approach it for the selected data set.

@lilyclements my understanding is that the selected data frame is displayed after applying the selection conditions on the original data frame, is this correct? if so since, currently, the  new column is added in the unselected data set, could we reload the selected data frame with the added columns included?

**Comment by lilyclements** [comment]
@Ivanluv yes you are right, and that's a good question/point. Personally, I'm not sure what is the best solution - I'm glad Shadrack fixed this bug though!

@Ivanluv @rdstern what do you both think is best? If you add a new column into a selected data frame, should it become a part of the selected data frame set of columns?
(I'm hoping this will be a trivial fix, but I am not as experienced in that bit of the R-Instat code)

**Comment by rdstern** [comment]
@Ivanluv @lilyclements I am hoping this issue has now been fixed.  The (at least partial) solution by @shadrackkibet is to remove the select when doing the calculation/transformation (I hope) and then to re-apply the select afterwards.  This has turned out to be quite a neat feature, as follows:
a) If the select is variables a, b, c. and then you make d <- a+b+c, then d is not included after making the calculation.
b) If the select is variables a, b, c & variables starting with d, then it will be included!
